backend/frontend/views.py

```diff
@@ -368,16 +377,63 @@ def action_start_card(request, card_id):
 
 
 @user_passes_test_or_forbidden(can_view_user_board)
-def course_component_details(request, project_id):
-    project = get_object_or_404(RecruitProject, id=project_id)
+def topic(request, id):
+    course_component = get_object_or_404(TopicProgress, id=id)
+
+    log_entries = LogEntry.objects.filter(object_1_id=id)
+
+    relevant_logs = []
+    for log_entry in log_entries:
+        event_type = EventType.objects.get(id=log_entry.event_type_id)
+
+        if (
+            course_component.agile_card.status == AgileCard.IN_PROGRESS
+            and event_type.name == CARD_STARTED
+        ) or (
+            course_component.agile_card.status == AgileCard.IN_PROGRESS
+            and event_type.name == CARD_REVIEW_REQUEST_CANCELLED
+        ):
+            relevant_logs.append(log_entry.timestamp)
+
+    formatted_time_difference = None
+    relevant_logs = sorted(relevant_logs)
+
+    if relevant_logs:
+        duration = timezone.now() - relevant_logs[-1]
+        seconds = duration.total_seconds()
+        days, remainder = divmod(seconds, 86400)
+        hours, remainder = divmod(remainder, 3600)
+        minutes, remainder = divmod(remainder, 60)
+
+        formatted_time_difference = (
+            f"{int(days)} days, {int(hours)} hours, {int(minutes)} minutes"
+        )
```

Comment:
@Ngoakor12, I agree with moving this to the TopicProgress model, but this will have to be a different method from the one you referenced. Our one calculates how long the card has been sitting in the current tilde column, while the one you referenced calculates the time it took for the card to move from start to completion.

Reply:
Ah okay, my bad. I see it in the task description now.